third_party/libusb/webport/src/libusb-proxy-receiver.js

```diff
@@ -62,7 +64,30 @@ GSC.LibusbProxyReceiver = class {
     /** @private @const */
     this.libusbChromeUsbBackend_ = libusbChromeUsbBackend;
     /** @private @const */
-    this.libusbToJsApiAdaptor_ = chooseLibusbToJsApiAdaptor();
+    this.realLibusbToJsApiAdaptor_ = chooseLibusbToJsApiAdaptor();
+    /** @type {!Array<!GSC.LibusbProxyHook>} @private @const */
+    this.hooks_ = [];
+  }
+
+  /**
+   * Adds a new hook. Multiple hooks are chained in the FILO order.
+   * @param {!GSC.LibusbProxyHook} hook
+   */
+  addHook(hook) {
+    // If it's the first hook, then chain it with the real implementation,
+    // otherwise - with the previously added hook.
+    hook.setDelegate(this.getLibusbToJsApiAdaptor_());
+    this.hooks_.push(hook);
+  }
+
+  /**
+   * @return {!GSC.LibusbToJsApiAdaptor}
+   * @private
+   */
+  getLibusbToJsApiAdaptor_() {
+    if (this.hooks_.length)
+      return this.hooks_[this.hooks_.length - 1];
```

Comment:
Starting with Chrome 92, we could also use this.hooks_.at(-1)
Though I guess we need to care about much older Chrome versions than that...

Reply:
Exactly - currently we claim the compatibility with Chrome >=48: https://github.com/GoogleChromeLabs/chromeos_smart_card_connector/blob/6d8ae8d0da3eac0b9f25ffb38b06d2a37649face/smart_card_connector_app/src/manifest.json.template#L29

Also if we were to drop it, jumping directly to 92 might be too aggressive, given a long tail of devices that stay on outdated OS versions...